x-pack/legacy/plugins/apm/public/hooks/useTransactionCharts.ts

```diff
@@ -7,29 +7,32 @@
 import { useMemo } from 'react';
 import { loadTransactionCharts } from '../services/rest/apm/transaction_groups';
 import { getTransactionCharts } from '../selectors/chartSelectors';
-import { IUrlParams } from '../context/UrlParamsContext/types';
-import { useUiFilters } from '../context/UrlParamsContext';
 import { useFetcher } from './useFetcher';
+import { useUrlParams } from './useUrlParams';
 
-export function useTransactionOverviewCharts(urlParams: IUrlParams) {
-  const { serviceName, start, end, transactionType } = urlParams;
-  const uiFilters = useUiFilters(urlParams);
+export function useTransactionCharts() {
+  const {
+    urlParams: { serviceName, transactionType, start, end, transactionName },
+    uiFilters
+  } = useUrlParams();
 
   const { data, error, status } = useFetcher(() => {
     if (serviceName && start && end) {
```

Comment:
`transactionName` used to be required here. I recall something about the overview chart being loaded when navigating to the details page when `transactionName` was not immediately available (or vice versa) which is way I separated those two. Might not be a problem anymore and then this simplification is good, just double checking.

Reply:
I thought about this, and then figured this should be pretty safe as the values come from the URL which is stable. Except... for redirects, I'll make sure that  check that.